### ISSUE TITLE
Windows 10 TaskBar - Notification/Action Center Icon Not Removable?

### ISSUE THREAD

**Issue by temartiniv** [valinet/ExplorerPatcher#817]
![image](https://user-images.githubusercontent.com/19539115/153009216-22ae584f-8abb-44ae-b400-4df6da6a5c28.png)

"Show Badges" is turned off, Windows 10 style taskbar is turned on. Even when there are no notifications, you get the "hollow" icon on the far right. Registry setting or GPO to remove entirely do not seem to work. Would prefer nothing to the right of the clock.

**Comment by chip33** [comment]
Hi , this should be a discussion not an issue as per  [DO NOT FILE ISSUES !!! USE DISCUSSIONS #56](https://github.com/valinet/ExplorerPatcher/issues/56)

but here is the solution.

![Screenshot 2022-02-08 150317](https://user-images.githubusercontent.com/48499740/153015703-767109e5-72a6-490f-a20c-9ae7d0652dfc.png)

![Screenshot 2022-02-08 150534](https://user-images.githubusercontent.com/48499740/153015634-8627ac66-4463-470a-9f90-7cbdd3e1a407.png)

**Comment by temartiniv** [comment]
Apologies, but this might more correctly qualify as an issue then.

1. When you are in Windows 10 TaskBar Mode, clicking "customize system icons in the notification area" immediately seems to initiate an explorer.exe restart, which closes the system icons window entirely before any action is possible
2. Switching to Windows 11 taskbar mode allows the window to remain open without triggering a restart, but Action Center is already turned off in the list

**Comment by valinet** [comment]
@temartiniv First of all, rules in https://github.com/valinet/ExplorerPatcher/issues/56 are there for a bunch of reasons that are already outlined there. As stated there, even though you consider it as an issue, the request is to still open it in Discussions so it can be properly sorted out. As for the topic at hand, it's actually at least partially a duplicate of an existing issue.

1. Is an old issue that's even present in Windows 10 (found it on latest LTSC 2021). It's an issue with `explorer.exe` itself, not ExplorerPatcher. There was a bug report filled a while ago: https://github.com/valinet/ExplorerPatcher/issues/236, but I never got to fixing it since it's not a bug in EP itself, I never figured out what causes it and it's very low priority since you're not spending much time in that window. To be successful in setting things there, wait for the clock to pass xx:00 minutes, and then you have a minute to set things in there. Go on, set stuff up, and before the clock reaches xx:59, click "Restart File Explorer" in EP. That will save the settings and `explorer` will restart with whatever options you have chosen and from there on it will work just fine. That Control Panel is anyway deprecated and that's probably the reason Microsoft never bothered to fix it up. I am not kidding, if you take a look at it, the crash always happens when the minute mark updates on the taskbar clock, so around xx:00, xx:01 etc.
2. That window shows options as configured for each of the taskbar modes: for the Windows 11 taskbar, Microsoft force disables those and instead they choose what to show in the system tray. For the Windows 10 taskbar, the options are available, but you have to hurry when changing the stuff in order to avoid the bug mentioned above.

**Comment by temartiniv** [comment]
@valinet Again, apologies for the issue creation.

I tried setting it at xx:00 of this hour and noticed the same behavior, and was unable to resolve. However, I was able to get some movement elsewhere. It does appear to be an issue with explorer and how it works with EP. I had seconds turned on my clock, and with seconds showing, the "customize system icons in the notification area" immediately triggers explorer reload when in Windows10 mode. If you turn seconds off, the reload occurs every time the clock updates (i.e: every time a new minute ticks over).

So, the fix is, turn off seconds, wait for a new minute to give yourself time, click "customize system icons in the notification area", set Action Center to Off, manually reload explorer.exe. If you do this you can then go back and turn seconds back on, and nothing resets the notification icons back to defaults.

I really appreciate the clarity you provided here. The only action that EP could make is to temporarily disable seconds when Windows10Mode and "customize system icons in the notification area" is clicked, allowing more time (whatever is left of the full minute on the clock) for the user to make the change. Either that, or pop up a message describing the clock-related problem with explorer.exe. I'm not saying either of those are your responsibility or even advisable from a development perspective. Thanks again.

**Comment by valinet** [comment]
Yeah, I am glad that at least I pointed you in the right direction. But as I mentioned, it's not necessarly a problem with ExplorerPatcher, but rather an issue the legacy shell has regarding that window and the taskbar clock. The reason I keep mentioning this is that you can reproduce the same bug on the recently released Windows 10 LTSC 2021 (based on 21H2) on which ExplorerPatcher does not work and which obviously ships with the Windows 10 UI. The bug happens there as well: just open that window, wait for the minute change mark, and `explorer.exe` will instantly crash and the desktop will reload.

It's a weird bug nevertheless, what's there to say...
